### ISSUE TITLE
invite people by link

### ISSUE THREAD

**Issue by albjeremias** [pixelfed/pixelfed#3815]
Yeah just like in mastodon:

![image](https://user-images.githubusercontent.com/26681446/203038981-2f5e715a-b936-4c10-bf51-dfd07ae79da4.png)

**Comment by mitexleo** [comment]
It's already available on Pixelfed

**Comment by mitexleo** [comment]
You have to set PF_USERS_INVITE= true on .env

**Comment by albjeremias** [comment]
it doesn't work..
https://github.com/pixelfed/pixelfed/issues/2238
also this you need to send an email. i want invitation links..

**Comment by okpierre** [comment]
Maybe this is what you are looking for? @mitexleo @albjeremias 

Invite Codes:

- Can be tied to a specific email address.
- Can be available to anyone (great for sharing on social media).
- Can have a limited number of uses or unlimited.
- Can have an expiry date, or never expire.

https://github.com/clarkeash/doorman

**Comment by mitexleo** [comment]
> Maybe this is what you are looking for? @mitexleo @albjeremias
> 
> Invite Codes:
> 
> * Can be tied to a specific email address.
> * Can be available to anyone (great for sharing on social media).
> * Can have a limited number of uses or unlimited.
> * Can have an expiry date, or never expire.
> 
> https://github.com/clarkeash/doorman

I'm aware of Doorman ... But not sure it will work with Pixelfed or not ..

**Comment by artburnGM** [comment]
In the same boat...
Email invites config is set up.
Option now available in each user's settings.
Attempted a few test invites.
User settings shows details of invites sent.
Yet, no invite going out or received by invitee.

Maybe I'm missing a config somewhere. Just figured it'd go out the same as all the other diagnosis files & new-user confirmation emails.

Or maybe this is still a "This feature has not been released yet."

And I'm with ya. Much rather be able to share an invite link (especially through DMs on other platforms).

**Comment by albjeremias** [comment]
> In the same boat... Email invites config is set up. Option now available in each user's settings. Attempted a few test invites. User settings shows details of invites sent. Yet, no invite going out or received by invitee.
> 
> Maybe I'm missing a config somewhere. Just figured it'd go out the same as all the other diagnosis files & new-user confirmation emails.
> 
> Or maybe this is still a "This feature has not been released yet."
> 
> And I'm with ya. Much rather be able to share an invite link (especially through DMs on other platforms).

no its not working. im managing by adding the users manually using 

`$ php artisan user:create username_here`

**Comment by artburnGM** [comment]
I was pondering doing that, but haven't tried it yet. (not planning on going over 50 users anytime soon. so it's not like it'd be hard to add 'em manually)

But I'd assume that if I add them manually, I'd have to know what they wanted for a username and of course their email before starting that process. Are they able to add their own password themselves and does it still do the email verification? 

Sounded like a better (although less "professional") idea until I think of things like this.

**Comment by albjeremias** [comment]
> I was pondering doing that, but haven't tried it yet. (not planning on going over 50 users anytime soon. so it's not like it'd be hard to add 'em manually)
> 
> But I'd assume that if I add them manually, I'd have to know what they wanted for a username and of course their email before starting that process. Are they able to add their own password themselves and does it still do the email verification?
> 
> Sounded like a better (although less "professional") idea until I think of things like this.

You are always welcome to create a bounty :) I am up for it... need money to buy bread! 
I don't have enough time to develop, cause i need to dumpsterdive to get ma foodz :P

https://sourceforge.net/software/product/Bountysource/alternatives

**Comment by stevenroose** [comment]
So the invite feature has been half-baked for at least 2.5 years. That's very interesting, yet unfortunate...

So there is absolutely no other way to run a private instance than creating usernames through command line?..

How about open registration but accounts requiring approval?

**Comment by dansup** [comment]
We do have a admin invite feature, and are working on user level invites!

**Comment by beedell-roke** [comment]
@dansup, does closure as a duplicate of that meta-issue mean that that effort has succeeded, or that it has been cancelled?
